db/seeds/api.model.js

```diff
@@ -0,0 +1,32 @@
+const db = require("../connection");
+const { topics, articles } = require("../data/test-data/index");
+console.log("In model");
+
+exports.selectTopics = (topics) => {
+  //console.log(topics, "<---model");
+  return db.query(`SELECT * FROM topics`).then(({ rows }) => {
+    return rows;
+  });
+};
+
+exports.checkArticleExists = (article_id) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+    .then(({ rows }) => {
+      if (!rows.length) {
+        return Promise.reject({ status: 404, msg: "Not Found" });
+      }
+    });
+};
+
+exports.selectArticle = (article_id) => {
+  let sqlQuery = `SELECT * FROM articles `;
+  const queryValues = [];
+  if (article_id) {
+    sqlQuery += `WHERE article_id = $1 `;
+    queryValues.push(article_id);
+  }
+  return db.query(sqlQuery, queryValues).then(({ rows }) => {
+    return rows;
+  });
+};
```

Comment:
Bit tricky with this one, I like what you're doing which is to build up an sqlQuery string based on what you'd like, this model is getting all your articles if there isn't a parameter and only one article if so, however, for a couple of reasons you should just call one model per controller, so one to get all articles and one to get an article based on ID. Reason 1 is that when you start building up queries in a future task, this model is going to have a lot of logic inside and therefore, if you do want to just get all articles, this model will check lots of things even when it doesn't need to. Reason 2 is, it can be confusing for people to read what is happening if there isn't a model per controller.